model.py

```diff
@@ -39,52 +21,118 @@ def __init__(
       self,
       name: str,
       provider: str = None,
-      # The JSON keys are in camelCase. To unpack these keys into
-      # Model attributes, we need to use the same camelCase names.
-      apiKey: str = None,  # pylint: disable=invalid-name
-      apiBase: str = None,  # pylint: disable=invalid-name
-      summarizeInstruction: str = None,  # pylint: disable=invalid-name
-      translateInstruction: str = None):  # pylint: disable=invalid-name
+      api_key: str = None,
+      api_base: str = None,
+      summarize_instruction: str = None,
+      translate_instruction: str = None,
+  ):
     self.name = name
     self.provider = provider
-    self.api_key = apiKey
-    self.api_base = apiBase
-    self.summarize_instruction = summarizeInstruction or DEFAULT_SUMMARIZE_INSTRUCTION  # pylint: disable=line-too-long
-    self.translate_instruction = translateInstruction or DEFAULT_TRANSLATE_INSTRUCTION  # pylint: disable=line-too-long
-
-  def completion(self, messages: List, max_tokens: float = None) -> str:
+    self.api_key = api_key
+    self.api_base = api_base
+    self.summarize_instruction = summarize_instruction or DEFAULT_SUMMARIZE_INSTRUCTION  # pylint: disable=line-too-long
+    self.translate_instruction = translate_instruction or DEFAULT_TRANSLATE_INSTRUCTION  # pylint: disable=line-too-long
+
+  def completion(self,
+                 instruction: str,
+                 prompt: str,
+                 max_tokens: float = None) -> str:
+    messages = [{
+        "role":
+            "system",
+        "content":
+            instruction + """
+Output following this JSON format:
+{"result": "your result here"}"""
```

Comment:
maybe `summary` or `translation`? instead of `result`

Reply:
I chose 'result' to simplify instructions and response parsing. If it's okay, I would like to keep it for now and update it more specifically if the quality of the response is poor